core/src/main/java/org/apache/iceberg/ManifestGroup.java

```diff
@@ -184,12 +192,14 @@ public <T extends ScanTask> CloseableIterable<T> plan(CreateTasksFunction<T> cre
                 specId -> {
                   PartitionSpec spec = specsById.get(specId);
                   ResidualEvaluator residuals = residualCache.get(specId);
-                  return new TaskContext(spec, deleteFiles, residuals, dropStats);
+                  return new TaskContext(spec, deleteFiles, residuals, dropStats, scanMetrics);
                 });
 
     Iterable<CloseableIterable<T>> tasks =
         entries(
             (manifest, entries) -> {
+              scanMetrics.totalDataFiles().increment(manifest.addedFilesCount());
+              scanMetrics.totalDeleteFiles().increment(manifest.deletedFilesCount());
```

Comment:
Delete files and deleted files are distinct. This is the number of deleted entries in the manifest, not the number of delete files.

In addition, existing data files will be used in the scan, not just added data files. Since this is counting inside of `ManifestGroup`, we need to make sure that all of the returned files are accounted for. There are two options:
1. Put these updates in `entries` where `ignoreDeleted` and `ignoreExisting` are checked
2. Add the total data files counter to the manifest reader, like you currently have for skipped files

I prefer option 2. I would opt to have simpler hooks instead of trying to count in bulk like this using metadata counts at the right place. That's probably going to be the most reliable way to get this data.

Reply:
ah that makes sense, thanks for double checking this. I added option 2 and I'm counting total data/delete files in the `ManifestReader`